packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/process.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class ChannelCommand extends FlutterCommand {
+  @override
+  final String name = 'channel';
+
+  @override
+  final String description = 'List or switch flutter channels.';
+
+  @override
+  String get invocation => '${runner.executableName} $name [<channel-name>]';
+
+  @override
+  bool get requiresProjectRoot => false;
+
+  @override
+  Future<int> runInProject() async {
+    switch (argResults.rest.length) {
+      case 0:
+        return await _listChannels();
+      case 1:
+        return await _switchChannel(argResults.rest[0]);
+      default:
+        printStatus('Too many arguments.');
+        printStatus(usage);
+        return 2;
+    }
+  }
+
+  Future<int> _listChannels() async {
+    String currentBranch;
+    await runCommandAndStreamOutput(
+      <String>['git', 'rev-parse', '--abbrev-ref', 'HEAD'],
+      workingDirectory: Cache.flutterRoot,
+      mapFunction: (String line) {
+        currentBranch = line;
+        return null;
+      },
+    );
+
+    printStatus('Flutter channels:');
+    return runCommandAndStreamOutput(
+      <String>['git', 'branch', '-r'],
+      workingDirectory: Cache.flutterRoot,
+      mapFunction: (String line) {
+        List<String> split = line.split('/');
+        if (split.length < 2) return null;
+        String branchName = split[1];
+        if (branchName.startsWith('HEAD')) return null;
+        if (branchName == currentBranch) return '* $branchName';
+        return '  $branchName';
```

Comment:
Deleted the spurious branches.

On Tue, Aug 23, 2016 at 7:14 PM, Eric Seidel <eseidel@google.com> wrote:

> I wouldn't bother adding an exclude list.  We'll just keep it clean.
>
> On Tue, Aug 23, 2016 at 7:11 PM, Dan Rubel <notifications@github.com>
> wrote:
>
>> In packages/flutter_tools/lib/src/commands/channel.dart
>> <https://github.com/flutter/flutter/pull/5561#discussion_r75984737>:
>>
>> > +        currentBranch = line;
>> > +        return null;
>> > +      },
>> > +    );
>> > +
>> > +    printStatus('Flutter channels:');
>> > +    return runCommandAndStreamOutput(
>> > +      <String>['git', 'branch', '-r'],
>> > +      workingDirectory: Cache.flutterRoot,
>> > +      mapFunction: (String line) {
>> > +        List<String> split = line.split('/');
>> > +        if (split.length < 2) return null;
>> > +        String branchName = split[1];
>> > +        if (branchName.startsWith('HEAD')) return null;
>> > +        if (branchName == currentBranch) return '* $branchName';
>> > +        return '  $branchName';
>>
>> Good point, but I would not want to hard code a fixed list into flutter.
>> Perhaps there should be a list of prefixes of branches that should be
>> ignored. At the moment, there are only 6 branches in the flutter repo...
>>
>> alpha
>> master
>> hackathon
>> revert-5426-build-sdk-summary
>> codelab
>> codelab2
>>
>> I think that hackathon* and codelab* could be (are?) valid channels that
>> we could use for planned events, but revert* should definitely be
>> excluded.
>>
>> We could ignore any branches that start with
>>
>>    - revert
>>    - test
>>    - any others?
>>
>> @sethladd <https://github.com/sethladd> Any thoughts on what branches
>> should be listed / excluded?
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/flutter/flutter/pull/5561/files/7126e2c9a00f14e9944aa9050ebb9e6de9aab311#r75984737>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ALTvi9S7X4PGjBK8ceJ7g_2vAEaG_iufks5qi6hagaJpZM4JrYW5>
>> .
>>
>
>